src/app/models/form.server.model.ts

```diff
@@ -718,17 +782,15 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     logicId: string,
     updatedLogic: LogicDto,
   ): Promise<IFormSchema | null> {
-    return this.findByIdAndUpdate(
-      formId,
-      {
-        $set: { 'form_logics.$[object]': updatedLogic },
-      },
-      {
-        arrayFilters: [{ 'object._id': logicId }],
-        new: true,
-        runValidators: true,
-      },
-    ).exec()
+    let form = await this.findById(formId).exec()
+    if (!form?.form_logics) return null
+    const index = form.form_logics.findIndex(
+      (logic) => logic._id.toHexString() === logicId,
```

Comment:
again I think it would be safer to use the `String` constructor instead of `toHexString`